examples/thumbs.rs

```diff
@@ -0,0 +1,69 @@
+//! Generate jpg thumbnails in parallel
+//!
+//! From <https://rust-lang-nursery.github.io/rust-cookbook/concurrency.html#generate-jpg-thumbnails-in-parallel>
+
+extern crate image;
+#[macro_use]
+extern crate quicli;
+extern crate structopt;
+
+use quicli::prelude::*;
+
+use std::path::{Path, PathBuf};
+use image::FilterType;
+
+/// Make some thumbnails
+#[derive(Debug, StructOpt)]
+struct Cli {
+    /// Which files?
+    #[structopt(default_value = "*.jpg")]
+    pattern: String,
+    /// How long should the longest edge of the thumbnail be?
+    #[structopt(long = "max-size", short = "s", default_value = "300")]
+    size: u32,
+    /// Where do you want to save the thumbnails?
+    #[structopt(long = "output", short = "o", default_value = "thumbnails")]
+    thumb_dir: String,
```

Comment:
~~Structopt can't be used for PathBuf~~ Actually just wanted to open an issue there and found <https://github.com/TeXitoi/structopt/issues/35>. `#[structopt(help = "Output filename", parse(from_os_str))]` is not very intuitive, though.